### PR TITLE
Add `target_kind` to scrubber matching rules

### PR DESCRIPTION
**Pull request by bozaro**
Allow the use of the target kind (for example, `cc_library`, `go_binary`, `go_test`...) in the scrubbing matcher configuration.
This is useful, as different target kinds can have the same mnemonic.

### CODE REVIEW

File: src/main/protobuf/remote_scrubbing.proto [hunk 1]
```diff
@@ -34,6 +34,10 @@ message Config {
     // Whether to match actions built for a tool configuration.
     // The default is to match only actions built for non-tool configurations.
     bool match_tools = 3;
+    // A regex matching the action target kind.
```
Comment (tjgq) [inline src/main/protobuf/remote_scrubbing.proto:37]: "A regex matching the rule kind of the action owner."

**Review by tjgq — CHANGES_REQUESTED** [review]
Looks good to me, but please add tests in `//src/test/java/com/google/devtools/build/lib/remote/ScrubberTest.java`. You can use the test cases for `label_regex` (i.e., `matchExactLabel`, `matchUnionLabel` and `matchWildcardLabel`) as a starting point.

**Review by tjgq — APPROVED** [review]

**Comment by tjgq** [comment]
I don't believe the CI failures are related. Importing.

**Comment by tjgq** [comment]
@bazel-io fork 7.1.0

**Comment by tjgq** [comment]
FYI - I renamed the `target_kind` config property to just `kind`, but otherwise the submitted code matches your PR.

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 7.1.0 RC1](https://releases.bazel.build/7.1.0/rc1/index.html). Please test out the release candidate and report any issues as soon as possible. If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=last_rc.
Thanks!

**Pull request closed without merging**